Code/Framework/AzToolsFramework/Tests/Prefab/Instance/InstanceDeserializationTests.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
+#include <AzToolsFramework/Prefab/PrefabDomUtils.h>
+#include <Prefab/PrefabTestFixture.h>
+
+namespace UnitTest
+{
+    using InstanceDeserializationTest = PrefabTestFixture;
+    using InstanceUniquePointer = AZStd::unique_ptr<AzToolsFramework::Prefab::Instance>;
+
+    static void GenerateDomAndReloadInstantiatedPrefab(InstanceUniquePointer& createdPrefab, InstanceUniquePointer& instantiatedPrefab)
+    {
+        PrefabDom tempDomForStoringAndLoading;
+        PrefabDomUtils::StoreInstanceInPrefabDom(*createdPrefab, tempDomForStoringAndLoading);
+
+        PrefabDomUtils::LoadInstanceFromPrefabDom(
+            *instantiatedPrefab, tempDomForStoringAndLoading, PrefabDomUtils::LoadFlags::UseSelectiveDeserialization);
+    }
+
+    AZStd::pair<InstanceUniquePointer, InstanceUniquePointer> SetupPrefabInstances(
+        const AzToolsFramework::EntityList& entitiesToUseForCreation, PrefabSystemComponent* prefabSystemComponent)
+    {
+        AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> createdPrefab =
+            prefabSystemComponent->CreatePrefab(entitiesToUseForCreation, {}, "test/path");
+        EXPECT_TRUE(createdPrefab != nullptr);
+
+        AZStd::unique_ptr<AzToolsFramework::Prefab::Instance> instantiatedPrefab =
+            prefabSystemComponent->InstantiatePrefab(createdPrefab->GetTemplateId());
+        EXPECT_TRUE(instantiatedPrefab != nullptr);
```

Comment:
repeat: this should use EXPECT_NE
```suggestion
        EXPECT_TRUE(nullptr, instantiatedPrefab);
```